psvpfsparser/FilesDbParser.cpp

```diff
@@ -348,6 +342,14 @@ bool FilesDbParser::parseFilesDb(std::ifstream& inputStream, std::vector<sce_ng_
          return false;
       }
 
+      //bad block with error or unknown format
+      bool is_bad_block = false;
+      if (block.header.nFiles > 9)
+      {
+         is_bad_block = true;
+         block.header.nFiles = 0;
```

Comment:
Here you set nFiles to 0. I am assuming by doing that you are treating file records in this block as unused data. Is that correct? They will be read with nUnusedSize1.

Reply:
I have seen only one case of this happening. With the changes I made, it decrypted successfully and matched exactly the decrypted data I got using vitashell.

I could not get it attached to this comment so I uploaded the files.db here:
https://mega.nz/#!SsEDXS4I!UJQWpquoXOYM1z6cY8aZCVCXNhIg7FMM5TEb3-iaozM

Reply:
Thanks for the provided information. This will require further investigation.
Please create a new clean issue and attach the file there.
We need to formulate the symptoms in the issue.
It looks like this unk6 == 0x400 and nFiles > 9 (which probably was -1 ?)
is some weird condition for block skipping. Just my guess.